### PR TITLE
[Components] kodagpt #11938

### PR DESCRIPTION
**Pull request by lcaresia**
## Summary by CodeRabbit

- **New Features**
  - Introduced semantic search functionality for the Kodagpt chatbot, allowing users to perform semantic searches within chatbot data.

- **Improvements**
  - Added new prop definitions and methods for interacting with the Kodagpt API to support the new semantic search feature.

### CODE REVIEW

File: components/kodagpt/kodagpt.app.mjs [hunk 1]
```diff
@@ -1,11 +1,43 @@
...
+          "Content-Type": "application/json",
+          "API_KEY": `${this.$auth.api_key}`,
+        },
+      });
+    },
+    async semanticSearch({
+      question, ...args
+    }) {
+      return this._makeRequest({
+        path: `/embed/${this.$auth.chatbot_id}/${question}`,
```
Comment (GTFalcao) [inline components/kodagpt/kodagpt.app.mjs:38]: Shouldn't `question` be URL-encoded here?

**Review by GTFalcao — COMMENTED** [review]
I'll move this forward to QA and I suggest testing with special characters in the question, as I believe it should be URL-encoded since it's being used directly in the path

**Review by luancazarine — APPROVED** [review]
Hi @lcaresia, thanks for your contribution! LGTM!

**Review by luancazarine — APPROVED** [review]

**Pull request merged**